From b656eaab4ce354effd844e0ec7f7f59bec9fe9c5 Mon Sep 17 00:00:00 2001
From: Nicolas Champciaux <nicolas.champciaux@intel.com>
Date: Tue, 18 Sep 2012 11:14:53 +0200
Subject: [PORT FROM MAIN] Revert "Revert "WLAN: Skip bind/unbind when no Wi-Fi module present""

BZ: 55966

This reverts commit 5da08b3495a31d8aa6885d0f2465e0e7b0433538.

Change-Id: I85686fa400ea8eeb740e1b479040a272cebd16c7
Orig-Change-Id: I5e9ca146223ef51f39e03f6d1cea5592415567af
Signed-off-by: Nicolas Champciaux <nicolas.champciaux@intel.com>
Reviewed-on: http://android.intel.com:8080/66887
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/77984
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wlan_prov/wlan_provisioning.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/wlan_prov/wlan_provisioning.c b/wlan_prov/wlan_provisioning.c
index a79513b..9531497 100644
--- a/wlan_prov/wlan_provisioning.c
+++ b/wlan_prov/wlan_provisioning.c
@@ -245,6 +245,11 @@ int main(int argc, char **argv)
 		return -1;
 	}
 
+       if (sdio_get_pci_id(SYSFS_SDIO_DEVICES_PATH, device_id)) {
+		LOGE("no wlan device detected, exit...");
+		return -1;
+       }
+
 #ifdef BUILD_WITH_CHAABI_SUPPORT
 	/* Read MAC address from Chaabi */
 	if (get_customer_data(ACD_WLAN_MAC_ADDR_FIELD_INDEX,
@@ -323,8 +328,7 @@ end:
 		* account. In that case, the reboot is required after flashing for the
 		* time the board.
 		*/
-		if (sdio_get_pci_id(SYSFS_SDIO_DEVICES_PATH, device_id)
-			|| unbind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)
+		if (unbind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)
 			|| bind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)) {
 			/*
 			* Rebooting the board: In this level, the NVS was saved. The
-- 
1.7.4.1

